meilisearch/src/search.rs

```diff
@@ -650,6 +694,22 @@ fn make_document(
     Ok(document)
 }
 
+/// Extract the JSON value under the field name specified
+/// but doesn't support nested objects.
+fn extract_field(
+    field_name: &str,
+    field_ids_map: &FieldsIdsMap,
+    obkv: obkv::KvReaderU16,
+) -> Result<Option<serde_json::Value>, MeilisearchHttpError> {
+    match field_ids_map.id(field_name) {
```

Comment:
nitpick, but couldn't the similarity be returned for each document by the underlying search, rather than being recomputed here by guessing that the value that was used to compute it was the one producing the highest similarity?

Reply:
It's much simpler to recompute this afterward and reduce the amount of code to inject and pass through layers between the search and the HTTP server.